lib/traject/bibtex_config.rb

```diff
@@ -17,71 +20,26 @@
-
-to_field 'book_title_ssim', from_bibtex(:booktitle)
-
-to_field 'pub_display', lambda { |record, accumulator, _context|
-  accumulator << record.journal.to_s.presence if record.respond_to?(:journal)
-  accumulator << record.publisher.to_s.presence if record.respond_to?(:publisher)
-}
-
-to_field 'location_ssi', from_bibtex(:address)
-
-to_field 'university_ssim', from_bibtex(:school)
-
-to_field 'edition_ssm', from_bibtex(:edition)
-
-to_field 'series_ssi', from_bibtex(:series)
-
-to_field 'thesis_type_ssm', from_bibtex(:type)
-
-to_field 'volume_ssm', from_bibtex(:volume)
-
-to_field 'issue_ssm', from_bibtex(:issue)
-
-to_field 'pages_ssm', from_bibtex(:pages)
-
-to_field 'doi_ssim', from_bibtex(:doi)
-
-to_field 'general_notes_ssim', lambda { |record, accumulator, _context|
-  accumulator << record.annote.to_s.presence if record.respond_to?(:annote)
-}
-
+to_field 'id', extract_bibtex_id
+to_field 'bibtex_key_ss', extract_bibtex_key
+to_field 'ref_type_ssm', extract_bibtex_field(:ref_type)
+to_fields %w(title_display title_full_display title_uniform_search title_sort), extract_bibtex_field(:title)
+to_fields %w(author_person_full_display author_sort), extract_bibtex_field(:author)
+to_fields %w(pub_year_isi pub_year_w_approx_isi), extract_bibtex_field(:year)
+to_field 'editor_ssim', extract_bibtex_field(:editor)
+to_field 'book_title_ssim', extract_bibtex_field(:booktitle)
+to_field 'pub_display', extract_bibtex_publication
+to_field 'location_ssi', extract_bibtex_field(:address)
+to_field 'university_ssim', extract_bibtex_field(:school)
+to_field 'edition_ssm', extract_bibtex_field(:edition)
+to_field 'series_ssi', extract_bibtex_field(:series)
+to_field 'thesis_type_ssm', extract_bibtex_field(:type)
+to_field 'volume_ssm', extract_bibtex_field(:volume)
+to_field 'issue_ssm', extract_bibtex_field(:issue)
+to_field 'pages_ssm', extract_bibtex_field(:pages)
+to_field 'doi_ssim', extract_bibtex_field(:doi)
+to_field 'general_notes_ssim', extract_bibtex_field(:annote)
 to_field 'format_main_ssim', literal('Reference')
-
-# raw serialization of BibTeX::Entry
-to_field 'bibtex_ts', from_bibtex(:raw)
-
-# formatted BibTeX::Entry in Chicago style as HTML
-to_field 'formatted_bibliography_ts', lambda { |record, accumulator, _context|
-  html = Bibliography.new(record.to_s).to_html
-  doc = Nokogiri::HTML(html)
-  li = doc.at_css('ol li')
-  reference = li.children.to_html if li.present? # extract just the reference from <li>
-  accumulator << reference.to_s
-}
-
-# Druids are kept as tags (keywords) in the BibTeX::Entry
-to_field 'related_document_id_ssim', lambda { |record, accumulator, _context|
-  record.keywords.to_s.split(',').map(&:strip).each do |druid|
-    accumulator << druid if druid =~ Exhibits::Application.config.druid_regex
-  end
-}
+to_field 'bibtex_ts', extract_bibtex_raw
+to_field 'formatted_bibliography_ts', extract_bibtex_formatted_bibliography
+to_field 'related_document_id_ssim', extract_bibtex_related_document_ids
```

Comment:
given rec below on adding traject options support to extract_bibtext_field(), this line would become: 

```ruby
to_field 'related_document_id_ssim', extract_bibtex_field(:keywords, split: ',', trim: true, match: Exhibits::Application.config.druid_regex)
``` 


which isn't as pretty in the config, but makes it easier to call on split, trim, match Extraction methods on a varying field mapping basis. This is something @camillevilla could use for the `annote` break later, for example.

Reply:
i moved that code into the bibtex macros